### PR TITLE
Adds a doc page on the ask_* methods

### PR DESCRIPTION
**Pull request by brandon-groundlight**

### CODE REVIEW

File: docs/docs/getting-started/6-ask-groundlight-methods.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+send the image to and return an `ImageQuery` object which contains the answer from Groundlight.
+The different methods provide different ways to utilize Groundlight's escalation logic.
+
+## `ask_ml`
+
+`ask_ml` is the most direct method for submitting an image query and asks for the fastest available ML answer
+from Groundlight. Groundlight's escalation logic will still run in order to improve the model's
+confidence for future queries.
+
+`ask_ml` is a good choice in most situations including live monitoring,
```
Comment (paulina-positronix) [inline docs/docs/getting-started/6-ask-groundlight-methods.md:14]:
```suggestion
`ask_ml` is a good choice in most situations including live monitoring.
```

File: docs/docs/getting-started/6-ask-groundlight-methods.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## `ask_confident`
+
+`ask_confident` give you access to high quality and human reviewed answers from Groundlight. Rather
+than get the first available answer, `ask_confident` will continue to wait as the image makes its
+way through the escalation process until either a more powerful model can give a high confidence
+answer or a human reviewer can provide a definitive answer. How confident the answer needs to be can
```
Comment (paulina-positronix) [inline docs/docs/getting-started/6-ask-groundlight-methods.md:21]:
```suggestion
way through the escalation process until either a more powerful model or a human reviewer can give a high confidence
answer. How confident the answer needs to be can
```

File: docs/docs/getting-started/6-ask-groundlight-methods.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+be configured through the `confidence_threshold` parameter.
+
+This is
+ideal in critical situations like quality control or safety monitoring where mistakes can be costly
+and you can wait a little longer for a more accurate answer. You can also use `ask_confident` as a
+filter after recieving an answer from `ask_ml` to provide a level of review before triggering any
+operations or contingencies.
+
+(You can program your own operations to be triggered by groundlight, but if you're interested in recieving text message or email notifications see if Groundlight alerts would meet your needs)
```
Comment (paulina-positronix) [inline docs/docs/getting-started/6-ask-groundlight-methods.md:30]: Consider including a link to documentation on GL alerts

File: src/groundlight/client.py [hunk 4]
```diff
@@ -408,35 +408,38 @@ def submit_image_query(  # noqa: PLR0913 # pylint: disable=too-many-arguments, t
...
-        :param wait: How long to poll (in seconds) for a confident answer. This is a client-side timeout.
+        :param wait: How long to poll (in seconds) for a confident answer. This is a client-side
+            timeout.
 
-        :param patience_time: How long to wait (in seconds) for a confident answer for this image query.
-            The longer the patience_time, the more likely Groundlight will arrive at a confident answer.
-            Within patience_time, Groundlight will update ML predictions based on stronger findings,
-            and, additionally, Groundlight will prioritize human review of the image query if necessary.
-            This is a soft server-side timeout. If not set, use the detector's patience_time.
+        :param patience_time: How long to wait (in seconds) for a confident answer for this image
```
Comment (brandon-groundlight) [inline src/groundlight/client.py:422]: I'm using the VSCode extension ReWrap to help make the line lengths of the docstrings more balanced. Most of the docstring changes are ReWrap

File: src/groundlight/client.py [hunk 5]
```diff
@@ -408,35 +408,38 @@ def submit_image_query(  # noqa: PLR0913 # pylint: disable=too-many-arguments, t
         :param detector: the Detector object, or string id of a detector like `det_12345`
 
         :param image: The image, in several possible formats:
+
```
Comment (brandon-groundlight) [inline src/groundlight/client.py:411]: The empty line is necessary in reStructuredText (otherwise the previous line is bolded in the sphinx docs)

Comment (paulina-positronix) [inline src/groundlight/client.py:411]: These are all linter changes, right?

**Review by paulina-positronix — COMMENTED** [review]
Would this go better under docs/building-applications/? There's already working with detectors and async queries entries there, and they might want to be updated with the right `ask_...` methods

**Pull request closed without merging**